src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -349,4 +356,19 @@ public ExecutionContext transform(Consumer<ExecutionContextBuilder> builderConsu
     public ResultNodesInfo getResultNodesInfo() {
         return resultNodesInfo;
     }
+
+    @Internal
+    public boolean isRunning() {
+        return isRunning.get() > 0;
+    }
+
+    @Internal
+    public void running() {
+        isRunning.incrementAndGet();
+    }
+
+    @Internal
+    public void finished() {
+        isRunning.decrementAndGet();
+    }
```

Comment:
Using a integer to represent "on and off" means you MUST get perfect code.  if something does 2 finishes and misses a .running() you will never be `isRunning()`

Why an integer ??